src/extract.rs

```diff
@@ -168,3 +168,38 @@ impl ToTokens for InstArguments {
         self.0.to_tokens(tokens)
     }
 }
+
+pub struct TestAttrs(pub(crate) Punctuated<NestedMeta, Token![,]>);
+
+impl TestAttrs {
+    pub fn try_extract(item: &mut ItemMod) -> syn::Result<Option<Self>> {
+        for (pos, attr) in item.attrs.iter().enumerate() {
+            if attr.path.is_ident("test_attributes") {
+                match attr.style {
+                    AttrStyle::Outer => {}
+                    AttrStyle::Inner(_) => {
+                        return Err(Error::new_spanned(attr, "cannot be an inner attribute"))
+                    }
+                };
+                let args = match attr.parse_meta()? {
+                    Meta::List(list) => list,
+                    Meta::Path(path) => {
+                        return Err(Error::new_spanned(
+                            path,
+                            "path syntax not supported; must apply at least one attribute",
+                        ))
+                    }
+                    Meta::NameValue(name_value) => {
+                        return Err(Error::new_spanned(
+                            name_value,
+                            "name-value syntax not supported",
+                        ))
+                    }
+                };
+                item.attrs.remove(pos);
+                return Ok(Some(TestAttrs(args.nested)));
+            }
```

Comment:
There is also the question of collecting across multiple `#[test_attributes(...)]`, but it would be cumbersome to implement cleanly, so we could punt on that for this first iteration 😅

Reply:
This is a good point. But yeah, I think I'm happy to leave that be for now.